components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -1397,13 +1397,29 @@ public void updateAPI(API api) throws APIManagementException, FaultGatewaysExcep
                             apiPublished.setOldOutSequence(oldApi.getOutSequence());
                             //old api contain what environments want to remove
                             Set<String> environmentsToRemove = new HashSet<String>(oldApi.getEnvironments());
-                            List<Label> labelsToRemove = new ArrayList<>(oldApi.getGatewayLabels());
+                            List<Label> labelsToRemove = null;
+                            if (oldApi.getGatewayLabels() != null){
+                                 labelsToRemove = new ArrayList<>(oldApi.getGatewayLabels());
+                            } else {
+                                 labelsToRemove = new ArrayList<>();
+                            }
+
                             //updated api contain what environments want to add
                             Set<String> environmentsToPublish = new HashSet<String>(apiPublished.getEnvironments());
-                            List<Label> labelsToPublish = new ArrayList<>(apiPublished.getGatewayLabels());
+                            List<Label> labelsToPublish;
+                            if (apiPublished.getGatewayLabels() != null ){
+                                labelsToPublish = new ArrayList<>(apiPublished.getGatewayLabels());
+                            } else {
+                                labelsToPublish = new ArrayList<>();
+                            }
 
                             Set<String> environmentsRemoved = new HashSet<String>(oldApi.getEnvironments());
-                            List<Label> labelsRemoved = new ArrayList<>(oldApi.getGatewayLabels());
+                            List<Label> labelsRemoved = null;
+                            if (oldApi.getGatewayLabels() != null){
```

Comment:
Can't we use the "if condition" at 1401 to this list too?

Reply:
https://github.com/1akshitha/carbon-apimgt/pull/4/commits/f877bc468fe1af638560a2eb2fc3ace19df0e508